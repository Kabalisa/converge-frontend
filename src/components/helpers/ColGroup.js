import React from 'react';
import Proptypes from 'prop-types';

const ColGroup = ({ numberOfMiddleColumns }) => (
  <colgroup>
    <col className="first-col" />
    {Array.from(Array(numberOfMiddleColumns).keys()).map(i => (
      <col key={`col${i}`} />
    ))}
    <col className="last-col" />
  </colgroup>
);

ColGroup.propTypes = {
  numberOfMiddleColumns: Proptypes.number,
};

ColGroup.defaultProps = {
  numberOfMiddleColumns: 2,
};

export default ColGroup;
