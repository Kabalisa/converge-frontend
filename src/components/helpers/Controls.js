import React from 'react';
import PropTypes from 'prop-types';
import '../../assets/styles/input.scss';
import chevron from '../../assets/images/chevron.svg';

const Controls = ({
  handleIncrement,
}) => (
  <div className="controls">
    <button className="control up" onClick={handleIncrement} name="up"><img src={chevron} alt="up" name="up" /></button>
    <button className="control down" onClick={handleIncrement} name="down"><img src={chevron} alt="down" name="down" /></button>
  </div>
);
Controls.propTypes = {
  handleIncrement: PropTypes.func.isRequired,
};
export default Controls;
