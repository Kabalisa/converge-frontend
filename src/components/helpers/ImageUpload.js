import toastr from 'toastr';
import * as firebase from 'firebase';
import notification from '../../utils/notification';
import config from '../../utils/FirebaseConfig';

firebase.initializeApp(config);

const getImageUrl = (folderName, imageFile) => {
  const storageRef = firebase.storage().ref(folderName + imageFile.name);

  return storageRef.put(imageFile)
    .then(() => (storageRef.getDownloadURL().then(url => url)))
    .catch(err => (notification(toastr, 'error', err)()));
};

export default getImageUrl;
