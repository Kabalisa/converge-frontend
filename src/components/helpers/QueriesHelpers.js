import { GET_USER_QUERY } from '../../graphql/queries/People';
import {
  GET_ALL_ROOMS,
  GET_LOCATIONS_QUERY,
  GET_ROOMS_QUERY,
  GET_ALL_REMOTE_ROOMS,
  GET_REMOTE_ROOMS_ALL_LOCATIONS,
} from '../../graphql/queries/Rooms';
import { GET_RESOURCES_QUERY } from '../../graphql/queries/Resources';
import ALL_ANALYTICS, { ANALYTICS_FOR_DAILY_ROOM_EVENTS } from '../../graphql/queries/analytics';
import { decodeTokenAndGetUserData } from '../../utils/Cookie';
import apolloClient from '../../utils/ApolloClient';
import GET_ALL_LEVELS from '../../graphql/queries/Levels';
import GET_ROOM_FEEDBACK_QUESTIONS_QUERY from '../../graphql/queries/questions';
import { GET_DEVICES_QUERY } from '../../graphql/queries/Devices';

const getUserDetails = async (client = apolloClient) => {
  const { UserInfo: userData } = decodeTokenAndGetUserData() || {};
  const email = process.env.NODE_ENV === 'test' ? 'converge@andela.com' : userData.email;
  try {
    const data = client.readQuery(
      {
        query: GET_USER_QUERY,
        variables: {
          email,
        },
      },
      true,
    );
    const user = Object.assign({}, data.user);
    user.firstName = userData.firstName;
    return user;
  } catch (err) {
    const { data } = await client.query({
      query: GET_USER_QUERY,
      variables: { email },
    });
    const user = Object.assign({}, data.user);
    user.firstName = userData.firstName;
    return user;
  }
};

const getAllLocations = async (client = apolloClient) => {
  try {
    const data = client.readQuery(
      {
        query: GET_LOCATIONS_QUERY,
      },
      true,
    );
    return data.allLocations;
  } catch (err) {
    const { data } = await client.query({
      query: GET_LOCATIONS_QUERY,
    });
    return data.allLocations;
  }
};

const getRoomList = async (userLocation, perPage, page, client = apolloClient) => {
  const { data } = await client.query({
    query: GET_ROOMS_QUERY,
    variables: {
      location: userLocation,
      office: '',
      page,
      perPage,
    },
  });
  return data;
};

const getAllResources = async (perPage, page, client = apolloClient) => {
  try {
    const data = client.readQuery(
      {
        query: GET_RESOURCES_QUERY,
        variables: {
          page: page || 1,
          perPage: perPage || 5,
        },
      },
      true,
    );
    return data.allResources;
  } catch (error) {
    const { data } = await client.query({
      query: GET_RESOURCES_QUERY,
      variables: {
        page: page || 1,
        perPage: perPage || 5,
      },
    });
    return data.allResources;
  }
};

const getAllRemoteRooms = async (client = apolloClient) => {
  const { data } = await client.query(
    {
      query: GET_ALL_REMOTE_ROOMS,
    },
    true,
  );
  return data.allRemoteRooms;
};

const getRemoteRoomsAllLocations = async (client = apolloClient) => {
  const { data } = await client.query(
    {
      query: GET_REMOTE_ROOMS_ALL_LOCATIONS,
      variables: {
        returnAll: true,
      },
    },
    true,
  );
  return data.allRemoteRooms;
};

const getRoomsStructure = async (client = apolloClient) => {
  const { data } = await client.query(
    {
      query: GET_ALL_LEVELS,
    },
    true,
  );
  return data;
};
const getRoomFeedbackQuestions = async (client = apolloClient) => {
  try {
    const data = client.readQuery(
      {
        query: GET_ROOM_FEEDBACK_QUESTIONS_QUERY,
      },
      true,
    );
    return data.questions;
  } catch (err) {
    const { data } = await client.query({
      query: GET_ROOM_FEEDBACK_QUESTIONS_QUERY,
    });
    return data.questions;
  }
};

const getAllRooms = async (location, client = apolloClient) => {
  try {
    const data = client.readQuery({
      query: GET_ALL_ROOMS,
      variables: {
        location,
      },
    }, true);
    return data;
  } catch (error) {
    const { data } = await client.query({
      query: GET_ALL_ROOMS,
      variables: {
        location,
      },
    }, true);
    return data;
  }
};

const getAllAnalytics = async (dateValue, client = apolloClient) => {
  try {
    const data = client.readQuery(
      {
        query: ALL_ANALYTICS,
        variables: dateValue,
      },
      true,
    );
    return data;
  } catch (err) {
    const { data } = await client.query({
      query: ALL_ANALYTICS,
      variables: dateValue,
    });
    return data;
  }
};

const query = (dateValue) => {
  const { startDate, endDate } = dateValue;
  return ({
    query: ANALYTICS_FOR_DAILY_ROOM_EVENTS,
    variables: {
      startDate,
      endDate,
    },
  });
};

const getAnalyticForDailyRoomsEvents
 = async (dateValue, client = apolloClient) => {
   try {
     const data = client.readQuery(
       query(dateValue),
       true,
     );
     return data;
   } catch (err) {
     const { data } = await client.query(
       query(dateValue),
     );
     return data;
   }
 };

const getAllDevices = async (client = apolloClient) => {
  try {
    const data = client.readQuery(
      {
        query: GET_DEVICES_QUERY,
      },
      true,
    );
    return data.allDevices;
  } catch (error) {
    const { data } = await client.query({
      query: GET_DEVICES_QUERY,
    });
    return data.allDevices;
  }
};

export {
  getAnalyticForDailyRoomsEvents,
  getAllAnalytics,
  getAllRooms,
  getUserDetails,
  getAllLocations,
  getAllResources,
  getAllRemoteRooms,
  getRoomList,
  getRoomsStructure,
  getRoomFeedbackQuestions,
  getRemoteRoomsAllLocations,
  getAllDevices,
};
