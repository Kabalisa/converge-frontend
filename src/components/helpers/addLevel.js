import Joi from 'joi-browser';

const AddLevel = (flattenedData, levelType) => {
  const Data = flattenedData.map((item) => {
    const { name } = item;
    return { name };
  });
  if (!levelType) {
    return 'Leveltype can not be empty';
  }

  const schema = {
    name: Joi.string().trim().min(2).required()
      .label(levelType),
  };
  const receivedErrors = Data.map((item) => {
    const { error: joiError } = Joi.validate(item, schema);
    if (joiError) {
      return joiError.details[0].message;
    }
    return '';
  });
  const receivedError = receivedErrors.find(item => item !== '');

  if (receivedError) {
    return receivedError;
  }
  return '';
};

export default AddLevel;
