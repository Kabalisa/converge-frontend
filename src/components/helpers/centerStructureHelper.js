/* eslint-disable no-await-in-loop */
/* eslint-disable no-restricted-syntax */
import {
  getRoomsStructure,
  getRoomId,
  getRoomResources,
} from './QueriesHelpers';

const isResourcesSetup = async (structureRooms) => {
  const roomIds = [];
  for (const room of structureRooms) {
    const roomId = await getRoomId(room.name);
    roomIds.push(roomId.data.getRoomByName[0]);
  }

  let allRoomsResources = [];

  for (const oneRoom of roomIds) {
    try {
      const oneRoomResource = await getRoomResources(oneRoom.id);
      allRoomsResources.push(oneRoomResource);
    } catch (error) {
      allRoomsResources = [[]];
    }
  }
  const levelResources = allRoomsResources
    .map(roomResources => roomResources.length > 0)
    .includes(true);
  return levelResources;
};
const Levels = (structure, level) =>
  structure.map(element => element.level === level).includes(true);
const LevelFilter = (structure, level) =>
  structure.filter(element => element.level === level);
const centerSetupLevel = async () => {
  const centerStructure = await getRoomsStructure();
  const levelOne = Levels(centerStructure.allStructures, 1);
  const levelFloors = Levels(centerStructure.allStructures, 2);
  const levelWings = Levels(centerStructure.allStructures, 3);
  const levelRooms = Levels(centerStructure.allStructures, 4);
  const structureRooms = LevelFilter(centerStructure.allStructures, 4);
  let levelResources;
  if (structureRooms.length) {
    levelResources = await isResourcesSetup(structureRooms);
  }
  // console.log('===========>levelResources', levelResources);
  if (!levelOne) {
    return 'welcomePage';
  }
  if (!levelFloors && !levelWings && !levelRooms && !levelResources) {
    return 'officeStructure';
  }
  if (!levelRooms && !levelResources) {
    return 'addRooms';
  }
  if (!levelResources) {
    return 'addResources';
  }
  return 'centerIsSetup';
};
export default centerSetupLevel;
