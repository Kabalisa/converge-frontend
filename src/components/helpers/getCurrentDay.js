
const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];

const newDay = () => (new Date()).getDate();
const newMonth = () => months[(new Date()).getMonth()];
const newYear = () => (new Date()).getFullYear();

const currentDate = () => (`${newMonth()} ${newDay()} ${newYear()}`);

export default currentDate;
