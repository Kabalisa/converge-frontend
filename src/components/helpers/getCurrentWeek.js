const getCurrentWeek = () => {
  const todayDate = new Date();
  const thisWeek = [];

  for (let i = 1; i <= 5; i += 1) {
    const first = (todayDate.getDate() - todayDate.getDay()) + i;
    const day = new Date(todayDate.setDate(first)).toDateString().slice(4, 15);
    thisWeek.push(day);
  }
  return thisWeek;
};

export default getCurrentWeek;
