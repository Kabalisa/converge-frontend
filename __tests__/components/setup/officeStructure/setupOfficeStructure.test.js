/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import uuid from 'uuid';

import SetupOfficeStructure from '../../../../src/components/setup/OfficeStructure/setupOfficeStructure';

jest.mock('../../../../src/components/helpers/QueriesHelpers');

import { getRoomsStructure } from '../../../../src/components/helpers/QueriesHelpers';

getRoomsStructure.mockImplementation(() => ({
  allStructures: [
    {
      id: '313',
      structureId: '7ddaeca4-d0b8-4675-9fa2-15ad68449cad',
      level: 1,
      name: 'X-Space',
      parentId: '',
      parentTitle: 'Kigali',
      tag: 'Buildings',
      locationId: 10,
      position: 1,
      state: 'StateType.active',
    }],
}));

describe('setupOfficeStructure Component', () => {
  const wrapper = shallow(<SetupOfficeStructure />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('calls the handleLevelTypeNumber method', () => {
    const instance = wrapper.instance();
    const { handleLevelTypeNumber } = instance;
    handleLevelTypeNumber('2');
    expect(wrapper.state().LevelTypeNumber).toEqual('2');
  });
  it('calls the handleLeveltype method', () => {
    const instance = wrapper.instance();
    const { handleLeveltype } = instance;
    const handleLeveltypeObj = {
      value: 'value',
    };
    handleLeveltype(null, handleLeveltypeObj);
    expect(wrapper.state().levelType).toEqual('value');
  });
  it('calls the handleLevelTypeName method default case', () => {
    const instance = wrapper.instance();
    const { handleLevelTypeName } = instance;
    const handleLevelTypeNameObj = {
      id: uuid(),
      value: 'value',
    };
    wrapper.setState({
      blocks: [{ structureId: uuid(), name: 'active' }],
      activeBlock: 'active',
      LevelTypeName: [
        {
          id: uuid(),
          name: 'name',
          buildingId: uuid(),
          buildingName: 'buildingName',
          inputTypeLevel: 'inputTypeLevel',
        },
      ],
      buildingLocationId: 100,
    });
    handleLevelTypeName(null, handleLevelTypeNameObj);
  });
  it('calls the handleLevelTypeName method initial case', () => {
    const instance = wrapper.instance();
    const { handleLevelTypeName } = instance;
    const handleLevelTypeNameObj = {
      id: uuid(),
      value: 'value',
    };
    wrapper.setState({
      blocks: [{ structureId: uuid(), name: 'active' }],
      activeBlock: 'active',
      LevelTypeName: [
        {
          id: '',
          name: '',
          buildingId: '',
          buildingName: '',
          inputTypeLevel: '',
        },
      ],
      buildingLocationId: 100,
    });
    handleLevelTypeName(null, handleLevelTypeNameObj);
  });
  it('calls the handleLevelTypeName method first case', () => {
    const instance = wrapper.instance();
    const { handleLevelTypeName } = instance;
    const commonId = uuid();
    const handleLevelTypeNameObj = {
      id: commonId,
      value: 'value',
    };
    wrapper.setState({
      blocks: [{ structureId: uuid(), name: 'active' }],
      activeBlock: 'active',
      LevelTypeName: [
        {
          id: commonId,
          name: 'name',
          buildingId: uuid(),
          buildingName: 'buildingName',
          inputTypeLevel: 'inputTypeLevel',
        },
      ],
      buildingLocationId: 100,
    });
    handleLevelTypeName(null, handleLevelTypeNameObj);
  });
  it('calls the toogleblock method', () => {
    const instance = wrapper.instance();
    const { toggleBlock } = instance;
    const block = 'anotherBlock';
    toggleBlock(block);
    expect(wrapper.state().activeBlock).toEqual('anotherBlock');
  });
});
