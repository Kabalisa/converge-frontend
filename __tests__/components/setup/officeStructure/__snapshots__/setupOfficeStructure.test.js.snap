// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setupOfficeStructure Component should render correctly 1`] = `
<SetupLayout
  layoutLeft={
    <UNDEFINED>
      <SetOfficeStructure
        activeBlock="X-Space"
        backIcon={
          <BackIcon
            handleOnClick={undefined}
            page="setupLocation"
          />
        }
        blocks={
          Array [
            Object {
              "name": "X-Space",
              "structureId": "7ddaeca4-d0b8-4675-9fa2-15ad68449cad",
            },
          ]
        }
        handleLeveltype={[Function]}
        levelType=""
        onInputChange={[Function]}
        toggleBlock={[Function]}
      />
      <NextButton
        flattenedData={
          Array [
            Object {
              "level": 2,
              "locationId": 10,
              "name": "",
              "parentId": "",
              "parentTitle": "",
              "position": 1,
              "structureId": "0fcdf19c-c397-4792-a1ef-b024d3e170fc",
              "tag": "",
            },
          ]
        }
        handleOnClick={undefined}
        levelType=""
      />
    </UNDEFINED>
  }
  layoutRight={
    <SetupLocationPreview
      content={
        <UNDEFINED>
          <OfficeStructurePreview />
        </UNDEFINED>
      }
    />
  }
/>
`;
